### PR TITLE
Bluetooth: define BT_COMPANY_ID_NORDIC

### PR DESCRIPTION
**Pull request by jori-nordic**
Also set BT_COMPANY_ID to BT_COMPANY_ID_NORDIC.

Making a kconfig option makes it also usable outside sdk-nrf.

### CODE REVIEW

**Review by tokangas — APPROVED** [review]
Ok for `modem_shell`.

**Review by alwa-nordic — APPROVED** [review]

**Review by KAGA164 — APPROVED** [review]
LGTM. I would consider changing  CONFIG_BT_COMPANY_ID_NORDIC->CONFIG_BT_COMPANY_ID in C language related files

File: samples/bluetooth/mesh/chat/include/chat_cli.h [hunk 1]
```diff
@@ -22,31 +22,28 @@ extern "C" {
...
 /* .. include_startingpoint_chat_cli_rst_1 */
-/** Company ID of the Bluetooth Mesh Chat Client model. */
-#define BT_MESH_CHAT_CLI_VENDOR_COMPANY_ID    0x0059
-
 /** Model ID of the Bluetooth Mesh Chat Client model. */
 #define BT_MESH_CHAT_CLI_VENDOR_MODEL_ID      0x000A
 
 /** Non-private message opcode. */
 #define BT_MESH_CHAT_CLI_OP_MESSAGE BT_MESH_MODEL_OP_3(0x0A, \
-				       BT_MESH_CHAT_CLI_VENDOR_COMPANY_ID)
```
Comment (omkar3141) [inline samples/bluetooth/mesh/chat/include/chat_cli.h:33]: I believe this should not be changed. We want to allow people to update/expand this model and use it for their own purpose and in that case, substitute this number with their own company ID. Applies to all changes in this file.

File: include/bluetooth/mesh/vnd/dm_common.h [hunk 2]
```diff
@@ -21,7 +21,6 @@
 extern "C" {
 #endif
 
-#define BT_MESH_VENDOR_COMPANY_ID 0x0059
```
Comment (omkar3141) [inline include/bluetooth/mesh/vnd/dm_common.h:24]: I believe this should not be changed. We want to allow people to update/expand this model and use it for their own purpose and in that case, substitute this number with their own company ID. Applies to all changes in this file.

File: subsys/bluetooth/mesh/shell/shell_dm_cli.c [hunk 3]
```diff
@@ -35,7 +35,7 @@ static int cmd_cfg(const struct shell *shell, size_t argc, char *argv[])
 		return err;
 	}
 
-	if (!mod && !shell_vnd_model_first_get(BT_MESH_VENDOR_COMPANY_ID,
```
Comment (omkar3141) [inline subsys/bluetooth/mesh/shell/shell_dm_cli.c:38]: I believe this should not be changed. We want to allow people to update/expand this model and use it for their own purpose and in that case, substitute this number with their own company ID. Applies to all changes in this file.

**Review by omkar3141 — CHANGES_REQUESTED** [review]

File: samples/bluetooth/mesh/chat/include/chat_cli.h [hunk 4]
```diff
@@ -23,7 +23,7 @@ extern "C" {
 
 /* .. include_startingpoint_chat_cli_rst_1 */
 /** Company ID of the Bluetooth Mesh Chat Client model. */
-#define BT_MESH_CHAT_CLI_VENDOR_COMPANY_ID    0x0059
+#define BT_MESH_CHAT_CLI_VENDOR_COMPANY_ID    CONFIG_BT_COMPANY_ID
```
Comment (omkar3141) [inline samples/bluetooth/mesh/chat/include/chat_cli.h:26]:
```suggestion
#define BT_MESH_CHAT_CLI_VENDOR_COMPANY_ID    CONFIG_BT_COMPANY_ID_NORDIC
```

File: samples/bluetooth/nrf_dm/src/main.c [hunk 5]
```diff
@@ -316,7 +315,7 @@ static int bt_sync_init(void)
 
 	printk("DM Bluetooth LE Synchronization initialization\n");
 
-	mfg_data.company_code = sys_cpu_to_le16(COMPANY_CODE);
+	mfg_data.company_code = sys_cpu_to_le16(CONFIG_BT_COMPANY_ID);
```
Comment (omkar3141) [inline samples/bluetooth/nrf_dm/src/main.c:318]: Isn't this Nordic Specific feature? I think this should be:
```suggestion
	mfg_data.company_code = sys_cpu_to_le16(BT_COMPANY_ID_NORDIC);
```

**Review by omkar3141 — CHANGES_REQUESTED** [review]

File: samples/bluetooth/nrf_dm/src/main.c (hunk 5 above)
Comment (jori-nordic) [reply samples/bluetooth/nrf_dm/src/main.c:318]: I don't know mesh that well :p

**Review by omkar3141 — APPROVED** [review]

**Pull request merged**